### PR TITLE
feat: add logger to extension

### PR DESCRIPTION
**Pull request by slavik-lvovsky**

### CODE REVIEW

File: packages/vscode-dependencies-validation/src/logger/logger.ts [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+ * Note the use of a getter function so the value would be lazy resolved on each use.
+ * This enables concise and simple consumption of the Logger throughout our extension.
+ */
+export function getLogger(): IVSCodeExtLogger {
+  return logger;
+}
+
+/* istanbul ignore next - ignoring "legacy" missing coverage to enforce all new code to be 100% */
+export function initLogger(context: ExtensionContext): void {
+  const extensionName = "vscode-dependency-validation"; // If the extension name changes, change this too
```
Comment (bd82) [inline packages/vscode-dependencies-validation/src/logger/logger.ts:21]: i think `context.extension.id` should give you the name

Comment (bd82) [reply packages/vscode-dependencies-validation/src/logger/logger.ts:21]: if not make a **test** to match this const value with what is in the pkg.json

Comment (slavik-lvovsky) [reply packages/vscode-dependencies-validation/src/logger/logger.ts:21]: we already have an output channel with this name.

File: packages/vscode-dependencies-validation/src/logger/logger.ts [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+/**
+ * Note the use of a getter function so the value would be lazy resolved on each use.
+ * This enables concise and simple consumption of the Logger throughout our extension.
+ */
+export function getLogger(): IVSCodeExtLogger {
+  return logger;
+}
+
+/* istanbul ignore next - ignoring "legacy" missing coverage to enforce all new code to be 100% */
```
Comment (bd82) [inline packages/vscode-dependencies-validation/src/logger/logger.ts:19]: the istanbul ignore comment here should be removed

File: packages/vscode-dependencies-validation/nyc.config.js [hunk 3]
```diff
@@ -3,5 +3,5 @@ module.exports = {
   // No integration or end to end tests currently in this project.
   // `extension.ts` only contains minimal integration code with VSCode.
   // and can be "safely" excluded.
-  exclude: ["src/extension.ts", "src/commands.ts"],
+  exclude: ["src/extension.ts", "src/commands.ts", "src/logger/logger.ts"],
```
Comment (bd82) [inline packages/vscode-dependencies-validation/nyc.config.js:6]: you may also need to add exclusions for the **root** nyc.config 
so it won't fail when it does the coverage merging

File: packages/vscode-dependencies-validation/src/logger/logger.ts [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+export function initLogger(
+  context: ExtensionContext,
+  outputChannel: OutputChannel
+): void {
+  const extensionName = context.extension.id; // If the extension name changes, change this too
```
Comment (bd82) [inline packages/vscode-dependencies-validation/src/logger/logger.ts:24]: the comment `// If the extension name changes, change this too` is not relevant anymore

File: packages/vscode-dependencies-validation/src/logger/logger.ts [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+  try {
+    logger = configureLogger({
+      extName: extensionName,
+      logPath: context.logPath,
+      logOutputChannel: outputChannel,
+      loggingLevelProp: LOGGING_LEVEL_CONFIG_PROP,
+      sourceLocationProp: SOURCE_TRACKING_CONFIG_PROP,
+      subscriptions: context.subscriptions,
+    });
+  } catch (error) /* istanbul ignore next -- this is complex to test and will give little value */ {
```
Comment (bd82) [inline packages/vscode-dependencies-validation/src/logger/logger.ts:34]: we are excluding this whole file from coverage so maybe the istanbul comment is redundant

**Review by bd82 — APPROVED** [review]
see minor comments

File: packages/vscode-dependencies-validation/src/logger/logger.ts (hunk 2 above)
Comment (slavik-lvovsky) [reply packages/vscode-dependencies-validation/src/logger/logger.ts:19]: Done.

File: packages/vscode-dependencies-validation/nyc.config.js (hunk 3 above)
Comment (slavik-lvovsky) [reply packages/vscode-dependencies-validation/nyc.config.js:6]: Done.

File: packages/vscode-dependencies-validation/src/logger/logger.ts (hunk 4 above)
Comment (slavik-lvovsky) [reply packages/vscode-dependencies-validation/src/logger/logger.ts:24]: Done.

File: packages/vscode-dependencies-validation/src/logger/logger.ts (hunk 5 above)
Comment (slavik-lvovsky) [reply packages/vscode-dependencies-validation/src/logger/logger.ts:34]: Done.

**Pull request merged**